utils/pg-introspection/src/acl.ts

```diff
@@ -134,80 +134,67 @@ export type ResolvedPermissions = Omit<AclObject, "role" | "granter">;
 const parseIdentifier = (str: string): string =>
   str.startsWith('"') ? str.replace(/"("?)/g, "$1") : str;
 
+// https://www.postgresql.org/docs/current/ddl-priv.html#PRIVILEGE-ABBREVS-TABLE
+const ACL_MAP = {
+  r: 'select',
+  w: 'update',
+  a: 'insert',
+  d: 'delete',
+  D: 'truncate',
+  x: 'references',
+  t: 'trigger',
+  X: 'execute',
+  U: 'usage',
+  C: 'create',
+  c: 'connect',
+  T: 'temporary',
+  m: 'maintain',
+};
+const NO_PERMISSIONS: AclObject = Object.values(ACL_MAP).reduce((acc, val) => {
+  acc[val] = acc[val + "Grant"] = false;
+  return acc;
+}, { role: "public", granter: "" });
+
 /**
  * Accepts an ACL string such as `foo=arwdDxt/bar` and converts it into
  * a parsed AclObject.
  */
 export function parseAcl(aclString: string): AclObject {
-  // https://www.postgresql.org/docs/current/ddl-priv.html#PRIVILEGE-ABBREVS-TABLE
-
-  const matches = aclString.match(/^([^=]*)=([rwadDxtXUCcTm*]*)\/([^=]+)$/);
-
-  if (!matches) {
+  const acl = Object.assign({}, NO_PERMISSIONS);
+  const roleEndIndex = aclString.indexOf("=");
+  if (roleEndIndex === -1) {
     throw new Error(`Could not parse ACL string '${aclString}'`);
+  } else if (roleEndIndex > 0) {
+    acl.role = parseIdentifier(aclString.substring(0, roleEndIndex));
+  }
+  const aclLength = aclString.length;
+  const lastGrantableTokenIndex = aclLength - 1;
+  let currentPerm: string;
```

Comment:
No need for this to be in a higher scope? In general I avoid `let` where possible.
```suggestion
```